script.js

```diff
@@ -0,0 +1,63 @@
+var Counter = React.createClass({
+    getInitialState: function () {
+        return {
+            counter: 0
+        };
+    },
+
+    increment: function () {
+        this.setState({
+            counter: this.state.counter + 1
+        })
+    },
+
+    decrement: function () {
+        this.setState({
+            counter: this.state.counter - 1
+        })
+    },
+
+    componentWillMount() {
+        console.log('Tutaj możemy wykonac operacje ale nie spowoduje ona re-renderowania')
+        return false;
+    },
+
+    shouldComponentUpdate() {
+        console.log('Tutaj możemy np. zatrzymać wszystkie kolejne metody cyklu życia zwracając false')
+        return this.state = true;
```

Comment:
Czemu ma służyć konstrukcja `this.state = true` ?

Reply:
wystarczy samo true ;) zmienione